### PR TITLE
details favoriting

### LINKED ISSUE #220: Add to favorites from river detail view in App
User suggestion: "I noticed you cannot favorite a run in map view, only in list view. It would be a nice addition to be able to add favorites from the run detail page."

### PR DESCRIPTION
**Pull request by yeahphil**
Fixes #220

- Add a favorite button to the run details right button bar items and wire it up
- Lol, DefaultsManager.userFavorites had no callers. Remove.
- Refactor: move reach detail label setting, etc to an updateDetailDisplay() func to reduce duplication
- Replace other condition color if/else blocks with conditionColor
- Light style/whitespace cleanup in RunDetailTableViewController

### CODE REVIEW

File: American Whitewater/Extensions/Reach.swift [hunk 1]
```diff
@@ -51,6 +51,16 @@ extension Reach {
             return ""
         }
     }
+    
+    /// Returns the 'runnability' color for a reaches condition
+    var conditionColor: UIColor {
```
Comment (Gregliest) [inline American Whitewater/Extensions/Reach.swift:56]: Lol, condition is such an ambiguous name. I guess we leave though for consistency.

Comment (Gregliest) [reply American Whitewater/Extensions/Reach.swift:56]: Actually, how hard would it to rename this to `runnabilityColor`?

Comment (yeahphil) [reply American Whitewater/Extensions/Reach.swift:56]: easy

Comment (yeahphil) [reply American Whitewater/Extensions/Reach.swift:56]: Or rather easy, but... I've been trying to refrain from renaming data model stuff, thinking the app is due for an effort to do that to most of the stuff in the models.

Doesn't seem like a problem to rename this as a one-off though, I'll go ahead and do it

Comment (Gregliest) [reply American Whitewater/Extensions/Reach.swift]: Yeah I think that's much better. Wasn't sure if it was used more widely.

**Pull request merged**